cql3/statements/select_statement.cc

```diff
@@ -462,8 +468,15 @@ select_statement::execute_without_checking_exception_message_aggregate_or_paged(
             state, options, command, std::move(key_ranges), _restrictions_need_filtering ? _restrictions : nullptr);
 
     if (aggregate || nonpaged_filtering) {
-        auto builder = cql3::selection::result_set_builder(*_selection, now, *_group_by_cell_indices);
-        coordinator_result<void> result_void = co_await utils::result_do_until([&p] {return p->is_exhausted();},
+        std::optional<uint64_t> limit;
+        if (_limit) {
+            limit = get_limit_impl(options, _limit);
```

Comment:
I still don't like this double-evaluation. We should evaluate the limit in one place, not several, and pass it around.

Reply:
Why "cache"? Evaluate the limit at the beginning and pass it around? If there something convoluted in the control flow?

I don't object to the double evaluation due to performance (this isn't a hot path), but due to cleanliness.